### PR TITLE
rgw: fix the two files with the same in the bucket when open/close version

### PR DESCRIPTION
**Pull request by chuang-he**
steps:
1.open version,  put the object.
2.close version, put the same object and set acl.
3.open version, put the same object.
4.list bucket

Signed-off-by: hechuang <hechuang@xsky.com>

### CODE REVIEW

**Comment by chuang-he** [comment]
test case https://github.com/ceph/s3-tests/pull/200

File: src/cls/rgw/cls_rgw.cc [hunk 1]
```diff
@@ -825,7 +825,7 @@ int rgw_bucket_complete_op(cls_method_context_t hctx, bufferlist *in, bufferlist
   }
 
   entry.index_ver = header.ver;
-  entry.flags = (entry.key.instance.empty() ? 0 : RGW_BUCKET_DIRENT_FLAG_VER); /* resetting entry flags, entry might have been previously a delete marker */
+  entry.flags = (entry.key.instance.empty() && !(entry.flags & RGW_BUCKET_DIRENT_FLAG_VER) ? 0 : RGW_BUCKET_DIRENT_FLAG_VER); /* resetting entry flags, entry might have been previously a delete marker */
```
Comment (ZVampirEM77) [inline src/cls/rgw/cls_rgw.cc]: please wrap these long line, how about
`/* resetting entry flags, entry might have been previously a delete marker */
entry.flags = (entry.key.instance.empty() &&
                       !(entry.flags & RGW_BUCKET_DIRENT_FLAG_VER) ? 0 : RGW_BUCKET_DIRENT_FLAG_VER);`
?

**Comment by chuang-he** [comment]
@ZVampirEM77 done

**Comment by mattbenjamin** [comment]
@yehudasa ?

**Pull request closed without merging**